### PR TITLE
[enhancements] Enhance developer challenge

### PR DESCRIPTION
**Pull request by dechdev**
- [x] Add hardhat for the solidity
- [x] Add a ERC-721 contract
- [x] Add mint button
- [x] Add smart contract addresses to config
- [x] Update README
- [x] Simple lint and build Github workflow

https://github.com/kaleido-io/developer-challenge/assets/17093033/308ba8a6-5155-4c70-b0ae-0ff4dbe9aa7a

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -45,6 +46,30 @@ function App() {
     setLoading(false);
   }
 
+  async function mintToken() {
+    setLoading(true);
+    setErrorMsg(null);
+    const randomTokenId = Math.floor(Math.random() * 1000);
```
Comment (lanasta) [inline frontend/src/App.tsx:52]: It's probably useful to note that token IDs must be unique and  this example would need to be enhanced for larger-scale minting to minimize confusion if the candidate does try to use this for their project

File: backend/src/index.ts [hunk 2]
```diff
@@ -1,36 +1,65 @@
...
-import simplestorage from "../contracts/simple_storage.json";
-import { v4 as uuidv4 } from "uuid";
+import express from "express";
+import simplestorage from "../../solidity/artifacts/contracts/simple_storage.sol/SimpleStorage.json";
+import token from "../../solidity/artifacts/contracts/token.sol/Token.json";
 
 const PORT = 4001;
 const HOST = "http://localhost:5000";
 const NAMESPACE = "default";
+const SIMPLE_STORAGE_ADDRESS = "ContractAddressHere";
```
Comment (shorsher) [inline backend/src/index.ts:10]: Should we add something to the README about ensuring they fill in this variable? or we could pull from an environment variable instead.

File: solidity/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Solidity Contracts: Kaleido Developer Challenge
+
+This folder demonstrates how to compile, test, and deploy two sample solidity contracts:
```
Comment (shorsher) [inline solidity/README.md]: maybe mention hardhat a bit more explicitly here and link to the docs?

**Review by shorsher — APPROVED** [review]

File: frontend/src/App.tsx (hunk 1 above)
Comment (dechdev) [reply frontend/src/App.tsx:52]: This is intentionally vague, and is a sample to be replaced, so I don't think we should elaborate here.

File: backend/src/index.ts (hunk 2 above)
Comment (dechdev) [reply backend/src/index.ts:10]: When the contracts compile, the console instructs them where to put the addresses. I think the candidate should be the one to productize and move it to an env variable.

**Review by peterbroadhurst — APPROVED** [review]

**Pull request merged**